app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,32 @@
+class PicturesController < ApplicationController
+
+  def index
+    @pictures = Picture.all
+  end
+
+  def show
+    @user = User.find(params[:user_id])
+    @picture = Picture.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @picture = @user.pictures.build
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @picture = Picture.new(picture_params)
```

Comment:
The `@picture.user = @user` line feels like model code.  Could the assembly of the picture be done in the model?